File: typescript/src/models/buyer.ts
// this file is @generated

export interface Buyer {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber?: string | null;
}

export const BuyerSerializer = {
  _fromJsonObject(object: any): Buyer {
    return {
      firstName: object["first_name"],
      lastName: object["last_name"],
      email: object["email"],
      phoneNumber: object["phone_number"],
    };
  },

  _toJsonObject(self: Buyer): any {
    return {
      first_name: self.firstName,
      last_name: self.lastName,
      email: self.email,
      phone_number: self.phoneNumber,
    };
  },
};

File: rust/src/models/buyer.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct Buyer {
    pub first_name: String,
    pub last_name: String,
    pub email: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub phone_number: Option<String>,
}
